### DISCUSSION TITLE
How do I update O3DE's Vulkan SDK?

### DISCUSSION THREAD

**Discussion by ohmaya** [o3de/o3de#12821] [Help Wanted]
GLAD?

**Comment by moudgils** [comment]
Glad vulkan header (generated here http://glad.sh/#api=vulkan%3D1.2&generator=c&options=HEADER_ONLY%2CLOADER%2CMX) is used to generate a loader and invoke Vulkan api calls for our needs. You only want to update the vulkan header for glad if you need to get access to some new api that is not available in the header at the moment. The header is located here - C:\GitHub\o3de\Gems\Atom\RHI\Vulkan\External\glad\2.0.0-beta\include\glad\vulkan.h. At run time we check physical device features to decide which version of vulkan (1.0, 1.1, 1.2) the device can support and switch to that version. So a feature like Ray Tracing is only enabled if the the device can support vulkan 1.2 at the very least. If you want to update to the latest vulkan sdk on you PC machine you can just download it from https://www.lunarg.com/vulkan-sdk/ and do it manually. If you find that you want to use a certain api in the new sdk that does not exist within the glad vulkan header you will need to re-generate the header.
